core/src/banking_stage/transaction_scheduler/greedy_scheduler.rs

```diff
@@ -0,0 +1,726 @@
+            match try_schedule_transaction(
+                transaction_state,
+                &pre_lock_filter,
+                &mut self.account_locks,
+                schedulable_threads,
+                |thread_set| {
+                    PrioGraphScheduler::<Tx>::select_thread(
+                        thread_set,
+                        &batches.total_cus,
+                        self.in_flight_tracker.cus_in_flight_per_thread(),
+                        &batches.transactions,
+                        self.in_flight_tracker.num_in_flight_per_thread(),
+                    )
+                },
+            ) {
+                Err(TransactionSchedulingError::Filtered) => {
+                    num_filtered_out += 1;
+                    container.remove_by_id(id.id);
+                }
+                Err(TransactionSchedulingError::UnschedulableConflicts) => {
+                    num_unschedulable += 1;
+                    self.unschedulables.push(id);
+                }
+                Ok(TransactionSchedulingInfo {
+                    thread_id,
+                    transaction,
+                    max_age,
+                    cost,
+                }) => {
+                    assert!(
+                        self.working_account_set.take_locks(&transaction),
+                        "locks must be available"
+                    );
+                    saturating_add_assign!(num_scheduled, 1);
+                    batches.transactions[thread_id].push(transaction);
+                    batches.ids[thread_id].push(id.id);
+                    batches.max_ages[thread_id].push(max_age);
+                    saturating_add_assign!(batches.total_cus[thread_id], cost);
+
+                    // If target batch size is reached, send all the batches
+                    if batches.ids[thread_id].len() >= self.config.target_transactions_per_batch {
+                        self.working_account_set.clear();
+                        num_sent += self.send_batches(&mut batches)?;
+                    }
+
+                    // if the thread is at target_cu_per_thread, remove it from the schedulable threads
+                    // if there are no more schedulable threads, stop scheduling.
+                    if self.in_flight_tracker.cus_in_flight_per_thread()[thread_id]
+                        + batches.total_cus[thread_id]
+                        >= target_cu_per_thread
+                    {
+                        schedulable_threads.remove(thread_id);
+                        if schedulable_threads.is_empty() {
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        self.working_account_set.clear();
+        self.send_batches(&mut batches)?;
+        assert_eq!(
+            num_scheduled, num_sent,
+            "number of scheduled and sent transactions must match"
+        );
```

Comment:
not sure why we need this assert. Also, shouldn't `num_sent` be incremented here?

Reply:
Yeah looks like CI just failed here

Reply:
`scheduled` means we put it into a batch.
`num_sent` assert is simply a sanity check to make sure we actually sent everything we scheduled to the workers, and didn't just drop them somewhere above.

Yes - forgot to increment num_sent here, which caused a test to fail (it should)

Fix: 98eaa985f8453b3c115ca38788f65a0a7d87d877